sdk/assertion_provider_pk.go

```diff
@@ -0,0 +1,145 @@
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+)
+
+// KeyAssertionID is the standard identifier for key-based assertions.
+//
+// Assertion ID Naming Convention:
+//   - System assertions (built-in): Use simple kebab-case names (e.g., "assertion-key", "system-metadata")
+//   - Custom assertions: Should use reverse-domain notation or descriptive prefixes to avoid conflicts
+//     (e.g., "com.example.custom-assertion", "app-specific-assertion")
+//
+// The simple "assertion-key" ID is used for SDK-provided key-based assertions and is part of the
+// standard TDF specification. Custom implementations should choose unique IDs that won't conflict
+// with current or future standard assertion types.
+const KeyAssertionID = "assertion-key"
+
+type PublicKeyStatement struct {
+	Algorithm string `json:"algorithm"`
+	Key       any    `json:"key"`
+}
+
+type KeyAssertionBinder struct {
+	privateKey AssertionKey
+	publicKey  AssertionKey
+}
+
+type KeyAssertionValidator struct {
+	publicKeys AssertionVerificationKeys
+}
+
+func NewKeyAssertionBinder(privateKey AssertionKey) *KeyAssertionBinder {
+	return &KeyAssertionBinder{
+		privateKey: privateKey,
+	}
+}
+
+func NewKeyAssertionValidator(publicKeys AssertionVerificationKeys) *KeyAssertionValidator {
+	return &KeyAssertionValidator{
+		publicKeys: publicKeys,
+	}
+}
+
+func (p KeyAssertionBinder) Bind(_ context.Context, m Manifest) (Assertion, error) {
+	// Create the public key statement
+	statement := PublicKeyStatement{
+		Algorithm: p.publicKey.Alg.String(),
+		Key:       p.publicKey.Key,
+	}
+
+	jsonBytes, err := json.Marshal(statement)
+	if err != nil {
+		return Assertion{}, fmt.Errorf("failed to marshal public key statement: %w", err)
+	}
+	statementValue := string(jsonBytes)
+
+	// Build the assertion
+	assertion := Assertion{
+		ID:             KeyAssertionID,
+		Type:           BaseAssertion,
+		Scope:          PayloadScope,
+		AppliesToState: Unencrypted,
+		Statement: Statement{
+			Format: StatementFormatJSON,
+			Schema: SystemMetadataSchemaV1,
+			Value:  statementValue,
+		},
+	}
+
+	// Get the hash and sign the assertion
+	assertionHash, err := assertion.GetHash()
+	if err != nil {
+		return assertion, fmt.Errorf("failed to get hash of assertion: %w", err)
+	}
+
+	// aggregation hash replaced with manifest root signature
+	if err := assertion.Sign(string(assertionHash), m.RootSignature.Signature, p.privateKey); err != nil {
+		return assertion, fmt.Errorf("failed to sign assertion: %w", err)
+	}
+
+	return assertion, nil
+}
+
+func (p KeyAssertionValidator) Verify(ctx context.Context, a Assertion, r Reader) error {
+	// Skip verification for assertions without bindings
+	if a.Binding.Signature == "" {
+		slog.WarnContext(ctx, "assertion has no binding, skipping verification",
+			slog.String("assertion_id", a.ID),
+			slog.String("assertion_type", string(a.Type)))
+		return nil
+	}
+
+	if p.publicKeys.IsEmpty() {
```

Comment:
This seems to also need to be influenced by the verification mode.  In FailFast or Strict I would expect an inability to verify resulting in a failure.